src/main/java/de/tum/in/www1/artemis/repository/specs/UserSpecs.java

```diff
@@ -102,6 +102,31 @@ public static Specification<User> getInternalOrExternalSpecification(boolean int
         }
     }
 
+    /**
+     * Creates the specification to look for the user without Registration Numbers.
+     *
+     * @param noRegistrationNumber true if the account should not have a registration number
+     * @param withRegistrationNumber true if the account should have a registration number
+     * @return specification used to chain database operations
+     */
+    public static Specification<User> getWithOrWithoutRegistrationNumberSpecification(Boolean noRegistrationNumber, Boolean withRegistrationNumber) {
+        if (!noRegistrationNumber && !withRegistrationNumber) {
+            return null;
+        }
+        else {
+            return (root, query, criteriaBuilder) -> {
+                if (noRegistrationNumber) {
+                    Predicate userWithoutRegistrationNumber = criteriaBuilder.isNull(root.get(User_.REGISTRATION_NUMBER));
```

Comment:
❓Question: I am not an expert on the criteria builder but withoutRegistrationNumber would mean for me null` or empty (meaning '') and withRegistrationNumber would mean **Not**(null or empty).

`criteriaBuilder.notEqual(root.get(User_.REGISTRATION_NUMBER), "");` does not seem to handle the null case?